admiral/component_envs/attacking.py

```diff
@@ -0,0 +1,83 @@
+
+from admiral.envs import Agent
+from admiral.component_envs.position import GridPositionAgent
+from admiral.component_envs.team import TeamAgent
+from admiral.component_envs.component import Component
+from admiral.component_envs.death_life import DyingAgent
+
+class GridAttackingAgent(Agent):
+    def __init__(self, attack_range=None, attack_strength=None, **kwargs):
+        assert attack_range is not None, "attack_range must be a nonnegative integer"
+        self.attack_range = attack_range
+        assert attack_strength is not None, "attack_strength must be a nonnegative number"
```

Comment:
Attack strength is not used here, but it is used by environments that process the agents. covered in #23